
Author: rmh
Status: minimal.c is normaly not compiled.
  As for version.c I'm not sure it really affects anything.

diff -ur procps-3.2.5.old/minimal.c procps-3.2.5/minimal.c
--- procps-3.2.5.old/minimal.c	2004-05-05 02:26:14.000000000 +0200
+++ procps-3.2.5/minimal.c	2005-10-10 18:15:05.000000000 +0200
@@ -57,7 +57,6 @@
 ///////////////////////////////////////////////////////
 #ifdef __linux__
 #include <asm/param.h>  /* HZ */
-#include <asm/page.h>   /* PAGE_SIZE */
 #define NO_TTY_VALUE DEV_ENCODE(0,0)
 #ifndef HZ
 #warning HZ not defined, assuming it is 100
@@ -67,9 +66,11 @@
 
 ///////////////////////////////////////////////////////////
 
-#ifndef PAGE_SIZE
-#warning PAGE_SIZE not defined, assuming it is 4096
-#define PAGE_SIZE 4096
+/* get PAGE_SIZE */
+#if defined(__linux__)
+# include <asm/page.h>
+#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+# include <machine/param.h>
 #endif
 
 
diff -ur procps-3.2.5.old/proc/version.c procps-3.2.5/proc/version.c
--- procps-3.2.5.old/proc/version.c	2003-01-29 02:11:43.000000000 +0100
+++ procps-3.2.5/proc/version.c	2005-10-10 18:13:46.000000000 +0200
@@ -35,6 +35,9 @@
 
 static void init_Linux_version(void) __attribute__((constructor));
 static void init_Linux_version(void) {
+#ifndef __linux__
+    int x = 2, y = 0, z = 0;
+#else
     static struct utsname uts;
     int x = 0, y = 0, z = 0;	/* cleared in case sscanf() < 3 */
     
@@ -45,5 +48,6 @@
 		"Non-standard uts for running kernel:\n"
 		"release %s=%d.%d.%d gives version code %d\n",
 		uts.release, x, y, z, LINUX_VERSION(x,y,z));
+#endif
     linux_version_code = LINUX_VERSION(x, y, z);
 }
